features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/APIDefinition/APIDefinition.jsx

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import PropTypes from "prop-types";
+import {withStyles} from "@material-ui/core/styles";
+import Button from "@material-ui/core/Button";
+import EditRounded from "@material-ui/icons/EditRounded";
+import CloudUploadRounded from "@material-ui/icons/CloudUploadRounded";
+import Dialog from "@material-ui/core/Dialog";
+import IconButton from "@material-ui/core/IconButton";
+import Icon from "@material-ui/core/Icon";
+import Paper from "@material-ui/core/Paper";
+import {FormattedMessage, injectIntl} from "react-intl";
+import ResourceNotFound from "../../../Base/Errors/ResourceNotFound";
+import {Progress} from "AppComponents/Shared";
+import Typography from "@material-ui/core/Typography";
+import Slide from "@material-ui/core/Slide";
+import MonacoEditor from "react-monaco-editor";
+import SwaggerEditorDrawer from "./SwaggerEditorDrawer";
+import yaml from "js-yaml";
+import Alert from "AppComponents/Shared/Alert";
+import Dropzone from "react-dropzone";
+
+const styles = theme => ({
+    titleWrapper: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+    },
+    swaggerEditorWrapper: {
+        height: '100vh',
+        overflowY: 'auto',
+    },
+    buttonIcon: {
+        marginRight: 10,
+    },
+    dropzone: {
+        border: 'none',
+        cursor: 'pointer',
+        padding: `${theme.spacing.unit * 2}px 0px`,
+        position: 'relative',
+        textAlign: 'center',
+    },
+});
+/**
+ * This component holds the functionality of viewing the api definition content of an api. The initial view is a
+ * read-only representation of the api definition file.
+ * Users can either edit the content by clicking the 'Edit' button or upload a new api definition file by clicking
+ * 'Import API Definition'.
+ * */
+class APIDefinition extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            openEditor: false,
+
+        };
+        this.api = props.api;
+        this.api_uuid = props.api.id;
```

Comment:
I think we don't need to assign the api id to an instance variable here, We could directly use the prop where we have used the instance variable `api_uuid` now ?

Reply:
You could access the props in  `componentDidMount` and get the api id directly from props

i:e

```javascript
const { api: { id } } = this.props;
```

This is nested object destructuring 
refer: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Nested_object_and_array_destructuring